streams/src/main/java/org/apache/kafka/streams/state/KeyValueStore.java

```diff
@@ -36,6 +37,7 @@
      * @param value The value to update, it can be {@code null};
      *              if the serialized bytes are also {@code null} it is interpreted as deletes
      * @throws NullPointerException If {@code null} is used for key.
+     * @throws InvalidStateStoreException if the store is not initialized
```

Comment:
Went through and added this additional javadoc line to the methods where it seemed to be missing. Looks like the usage throughout the codebase is very inconsistent though:
* `RocksDBStore` enforces it on all methods even though it wasn't documented in `KeyValueStore`. The in-memory implementations do not enforce it.
* `WindowStore` has docs about this but the actual implementations do not enforce it.
* `VersionedKeyValueStore` has docs but the implementation was not enforcing it. I've reconciled this inconsistency in this PR, but the others are larger changes that I'd like to leave for later.